File: NirvanaSMS.Api.CSharp/Core/Utility/ManyToManyMessageFormatter.cs
using NirvanaSMS.Api.Core.Base;
using NirvanaSMS.Api.Core.Entity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NirvanaSMS.Api.Core.Utility
{
    public class ManyToManyMessageFormatter : FormatterBase<ManyToManyMessage, MessageBase>
    {
        public override ManyToManyMessage Format(MessageBase messages)
        {
            PhoneNumberFormatter phoneNumberFormatter = new PhoneNumberFormatter();
            ManyToManyMessage _messages = messages as ManyToManyMessage;
            foreach (var item in _messages.PhonesAndMessages)
            {
                phoneNumberFormatter.Format(item.Number);
                //item.Message.Trim();
            }
            return _messages;
        }
    }
}

File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/core/utility/ManyToManyMessageFormatter.java
package org.nirvanasms.api.core.utility;

import org.nirvanasms.api.core.base.FormatterBase;
import org.nirvanasms.api.core.base.MessageBase;
import org.nirvanasms.api.core.entity.ManyToManyMessage;
import org.nirvanasms.api.core.entity.PhoneAndMessagePair;

public class ManyToManyMessageFormatter extends FormatterBase<ManyToManyMessage, MessageBase> {
	@Override
	public ManyToManyMessage format(MessageBase messages) {
		PhoneNumberFormatter phoneNumberFormatter = new PhoneNumberFormatter();
		ManyToManyMessage _messages = (ManyToManyMessage) messages;
		for (PhoneAndMessagePair item : _messages.getPhonesAndMessages()) {
			phoneNumberFormatter.format(item.getNumber());
		}
		return _messages;
	}
}
